salt/states/rsync.py

```diff
@@ -97,7 +99,10 @@ def synchronized(name, source,
     '''
     ret = {'name': name, 'changes': {}, 'result': True, 'comment': ''}
 
-    if not os.path.exists(source):
+    if name.strip() == '/':
```

Comment:
Understood and I agree that this is likely to cause problems but I'm not of the opinion that it's Salt's role to prevent these types of issues. If you wouldn't mind removing this, I will merge this after it's gone. Thanks.